### PR TITLE
Documentation || Extending Avo::BaseResource

### PR DESCRIPTION
**Pull request by Yash-Singh-Pathania**
Description for extending Avo::BaseResource allowing customization 
feature pr for the same can be found here : [pull request](https://github.com/avo-hq/avo/pull/2998)

### CODE REVIEW

File: docs/3.0/resources.md [hunk 1]
```diff
@@ -367,6 +367,50 @@ end
...
+
+      super(id, **args, &block)
+    end
+  end
+end
+```
+
+## Using Your Custom `Avo::BaseResource`
+
+All your resources will automatically inherit from your custom `Avo::BaseResource`, allowing you to add common functionality across your admin interface. For instance, the above example ensures that all number fields in your resources will have their values cast to floats. You can add any other shared methods or customizations here, making it easier to maintain consistent behavior across all resources.
```
Comment (adrianthedev) [inline docs/3.0/resources.md:396]:
```suggestion

All your resources will now inherit from your custom `Avo::BaseResource`, allowing you to add common functionality across your admin interface. For instance, the above example ensures that all number fields in your resources will have their values cast to floats. You can add any other shared methods or customizations here, making it easier to maintain consistent behavior across all resources.
```

File: docs/3.0/resources.md [hunk 2]
```diff
@@ -367,6 +367,50 @@ end
...
+
+## Example Resource
```
Comment (adrianthedev) [inline docs/3.0/resources.md:398]:
```suggestion
### Your resource files
```

File: docs/3.0/resources.md [hunk 3]
```diff
@@ -367,6 +367,50 @@ end
...
+module Avo
+  module Resources
+    class Post < Avo::BaseResource
+      # Your existing configuration for the Post resource
+    end
+  end
+end
+```
+
+By following this approach, you can leverage the full power of `Avo::BaseResource` keeping it  maintainable, while also providing flexibility for future customizations.
```
Comment (adrianthedev) [inline docs/3.0/resources.md:413]:
```suggestion
Your resource file will still look the same as it did before.

```ruby
# app/avo/resources/post_resource.rb
module Avo::Resources::Post < Avo::BaseResource
  # Your existing configuration for the Post resource
end
```

By following this approach, you can leverage the full power of `Avo::BaseResource` keeping it  maintainable, while also providing flexibility for future customizations.
```

**Review by adrianthedev — APPROVED** [review]
Great work @Yash-Singh-Pathania! Thank you for the docs and the feature!

File: docs/3.0/resources.md [hunk 4]
```diff
@@ -367,6 +367,42 @@ end
 This tells Avo which resources you use and stops the eager-loading process on boot-time.
 This means that other resources that are not declared in this array will not show up in your app.
 
+## Extending `Avo::BaseResource`
+
+We have restructured the `Avo::BaseResource` to enhance user customization capabilities. The existing functionality has been moved to a new base class `Avo::Resources::Base`, and `Avo::BaseResource` is now left empty for user overrides. This allows users to easily add custom methods that all of their resources will inherit, without having to modify the internal base class.
```
Comment (Paul-Bob) [inline docs/3.0/resources.md:372]:
```suggestion
<VersionReq version="3.10.7" /> we have restructured the `Avo::BaseResource` to enhance user customization capabilities. The existing functionality has been moved to a new base class `Avo::Resources::Base`, and `Avo::BaseResource` is now left empty for user overrides. This allows users to easily add custom methods that all of their resources will inherit, without having to modify the internal base class.
```

**Review by Paul-Bob — APPROVED** [review]
Thanks @Yash-Singh-Pathania it's looking great.

I added the information for the version requirement

**Pull request merged**